### ISSUE TITLE
Build a conversational faction dataset and held-out identity benchmark

### ISSUE THREAD

**Issue by KelianM** [KelianM/endless-voices#1]
## Goal

Build the first reviewed conversational dataset and a separate held-out benchmark for Endless Voices. The README's central question is whether LoRA fine-tuning improves fictional identity fidelity compared with prompting alone. Teach lore, worldview, behaviour, and speaking style through dialogue—not raw game-text dumps—and measure factual accuracy separately from sounding distinctive.

The current repository has a JSONL loader, LoRA training, and interactive chat, but only one original, non-canonical archivist example and no evaluation framework. This issue covers the dataset and evaluation foundation; demonstrating a fine-tuning improvement is a later experiment, not an acceptance requirement.

## Dataset scope and authoring

- Select and document a small initial set of Endless Sky species/factions. Record stable identity IDs, species/faction relationships, character roles, and the game/source revision used as canon.
- Create reviewed identity briefs covering history, knowledge boundaries, values, attitudes toward humans/other species, behaviour, and speech patterns. Distinguish canonical facts, character opinions, and plausible creative dialogue.
- Write original conversational examples across history, interspecies relations, moral decisions, technology, conflict, and everyday interaction. Include follow-up conversations and unfamiliar situations that require applying values rather than repeating facts.
- Track source references, attribution/permissions, authorship or generation method, and review status. Review generated examples against canon before inclusion. Do not treat the existing invented archivist as canonical data.
- Publish a dataset card with coverage counts by identity/topic, known gaps, source policy, and limitations. Choose and record pilot counts before authoring; keep coverage reasonably balanced.

## Training example contract

Preserve the existing UTF-8 JSONL contract: one complete conversation per line, optional leading system message, then one or more complete alternating user/assistant pairs. For this dataset, require an identity-setting system message and metadata, even though the existing loader permits their omission.

Illustrative schema only; placeholders below are not training content:

```json
{
  "messages": [
    {
      "role": "system",
      "content": "You are a representative of <faction>, a member of <species>. <Character role and identity/style instruction.>"
    },
    {
      "role": "user",
      "content": "A human trader asks for shelter while an enemy patrol approaches. What do you do?"
    },
    {
      "role": "assistant",
      "content": "<Reviewed in-character response demonstrating the faction's values, behaviour, and speaking style.>"
    }
  ],
  "metadata": {
    "id": "example-0001",
    "identity": "<stable-faction-id>",
    "species": "<stable-species-id>",
    "character_role": "<role>",
    "topics": ["moral_decision", "humans", "conflict"],
    "scenario_group": "shelter-enemy-patrol",
    "split": "train",
    "sources": [
      {"reference": "<source path or URL and passage>", "revision": "<pinned revision>"}
    ],
    "authorship": "human",
    "review_status": "approved"
  }
}
```

Metadata is for curation and evaluation, not model input; the loader currently ignores it. Add separate dataset validation for required fields, IDs, sources, review state, split membership, and coverage. Validate messages with the existing loader and check token lengths using the selected tokenizer; do not silently truncate responses. Document that the current trainer computes loss on all non-padding tokens, including system/user text. Assistant-only loss is outside this issue.

## Split policy and benchmark format

- Maintain physically separate training, development, and frozen benchmark files with versioned manifests/checksums.
- Split by complete conversation and scenario family **before** generating variations. Keep paraphrases, shared dialogue templates, and derived examples in the same split; run exact and near-duplicate checks plus manual review.
- Use development data for prompt design, hyperparameters, and checkpoint selection. Never train on benchmark prompts/responses or use benchmark results to tune prompts or select checkpoints. If iterative development uses a released benchmark, disclose that and establish a fresh final holdout.
- Shared canonical facts are legitimate across splits: hold out the conversational tasks/scenarios, with a dedicated unseen-scenario subset. Do not claim these tests establish absence of contamination in the base model's pretraining.
- Store benchmark cases separately from training conversations: case ID, identity, topic, scenario group, user turn(s), applicable scoring dimensions, source-backed expected facts/behaviours/style indicators, prohibited contradictions, and uncertainty expectations. Reference answers may guide judges but are not exact-match targets.
- Keep references, scoring notes, and expected answers out of model inputs. For multi-turn cases, use fixed user follow-ups and each model's own generated history, not gold assistant answers.
- Include factual recall, worldview/behaviour decisions, speech style, novel scenarios, false-premise/unknown-fact probes, and identity persistence across multiple turns.

## Benchmark scoring criteria

Score each applicable dimension independently from **0–4**, supported by response excerpts and source references:

| Dimension | What to assess |
| --- | --- |
| Lore consistency | Correctness and coverage of relevant, source-backed history, relationships, and knowledge boundaries. |
| Personality, worldview, and behaviour | Whether priorities, judgments, and proposed actions fit the identity and remain coherent across turns. |
| Speaking style | Appropriate vocabulary, register, phrasing, and conversational manner; distinctive style alone does not earn lore credit. |
| In-character generalization | Whether unfamiliar situations are handled coherently by applying the identity's values without abandoning the role. |
| Factual restraint | Avoiding unsupported canonical claims and contradictions; handling unknown facts and false premises appropriately in-character. Harmless creative details are not automatically errors. |

Shared anchors: **0** = fundamentally wrong, contradictory, or absent; **1** = major failures; **2** = mixed/partially correct with meaningful lapses; **3** = meets expectations with minor lapses; **4** = fully meets the case-specific criteria with no material lapse. Add annotated examples for each dimension to calibrate these anchors.

Mark non-applicable dimensions in advance. Report per-dimension scores, per-identity/topic breakdowns, and a macro-average across identities (equal-weight applicable dimensions within each case). Separately report the percentage of responses containing a source-backed contradiction and multi-turn identity breaks, so style scores cannot hide factual failures. Do not reward evasiveness: failure to answer known lore questions lowers lore coverage even if no facts are invented.

Blind and shuffle condition labels during review. Have two reviewers independently score a documented subset, record agreement/disagreements, and adjudicate material differences. Manual scoring is sufficient for the pilot; any later automated judge must be calibrated against reviewed examples and have its model/prompt/version recorded.

## Reproducible comparison protocol

Use the same starting checkpoint/revision, tokenizer, benchmark cases, context budget, and decoding settings across conditions:

1. **Base:** starting checkpoint (possibly already instruction-tuned), no identity system prompt or adapter.
2. **Prompted:** same checkpoint with a frozen, carefully written identity prompt.
3. **Fine-tuned + prompt:** same checkpoint plus the LoRA adapter and exactly the same identity prompt as condition 2. This is the primary comparison for incremental fine-tuning benefit.
4. **Adapter-only diagnostic:** where identity selection is unambiguous, also evaluate an adapter without the rich identity prompt, as suggested by the README. Document faction-specific versus shared-adapter routing; do not silently inject identity context into the base condition.

Identity is evaluator metadata in the no-context baseline; report that this condition measures behavior without identity selection. Freeze prompts using development data only. Record dataset/benchmark versions, adapter identity, training config, dependencies, seeds, rendered prompts, outputs, scores, and decoding settings. Use greedy decoding initially or identical documented seeds/repeats when sampling. Report paired differences on the same cases and sample counts; training loss is not a benchmark score.

## Acceptance criteria

- [ ] Reviewed pilot dataset, identity briefs, provenance records, and dataset card are present.
- [ ] Training schema and authoring guidance are documented with valid examples and validators.
- [ ] Train/development/benchmark manifests, coverage counts, and leakage checks are recorded.
- [ ] Frozen benchmark covers all five dimensions, unseen scenarios, and multi-turn persistence.
- [ ] Case-specific scoring notes, anchored rubric, scoring template, and reviewer calibration examples are included.
- [ ] A minimal evaluation command generates and saves responses for the defined conditions; a documented scoring workflow produces dimension scores, breakdowns, and failure rates.
- [ ] Offline checks cover schema errors, split leakage, and exclusion of benchmark references from model inputs; a smoke evaluation verifies output/report structure.
- [ ] README explains how to build/validate the data and run the comparison, clearly separating pilot validation from model-quality results.

Large-scale synthetic generation, a frontend, and full training sweeps are outside this issue.
